### PR TITLE
Add SIG API Machinery spotlight

### PR DESCRIPTION
**Pull request by fsmunoz**
Add SIG API Machinery spotlight interview.

Fixes: #469

### CODE REVIEW

**Comment by fsmunoz** [comment]
I'm still reviewing and adding links to it, will remove the WIP tag as soon as I have it ready for additional reviews.

File: content/en/blog/2024/sig-api-machinery-spotlight.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+---
+
+We recently talked with [Federico Bongiovanni](https://github.com/fedebongio) (Google) and [David
+Eads](https://github.com/deads2k) (Red Hat), Chairs of SIG API Machinery, to know a bit more about
+this Kubernetes Special Interest Group.
+
+## Introductions
+
+**Frederico (FSM)**: Hello, and thank your time. To start with, could you tell us about
+yourselvesand how you got involved in Kubernetes?
```
Comment (kaslin) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:16]: *yourselves and

File: content/en/blog/2024/sig-api-machinery-spotlight.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+**David**: I started working on
+[OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift) (the Red Hat
+distribution of Kubernetes) in the fall of 2014 and got involved pretty quickly in apimachinery.  My
```
Comment (kaslin) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:20]: *apimachinery -> API Machinery

File: content/en/blog/2024/sig-api-machinery-spotlight.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+activity.
+
+1. **Finding purpose**: `pre-1.0` up until `v1.3` (up to our first 1000+ nodes/namespaces) or
+   so. This time was characterized by rapid change.  We went through five different versions of our
+   schemas and rose to meet the need.  We optimized for quick, in-tree API evolution (sometimes to
+   the detriment of longer term goals), and defined patterns for the first time.
+
+2. **Scaling to meet the need**: `v1.3-1.9` (up to shared informers in controllers) or so.  When we
+   started trying to meet customer needs as we gained adoption, we found severe scale limitations in
+   terms of CPU and memory. This where we broadened API machinery to include access patterns, but
```
Comment (kaslin) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:112]: *This where -> This was where

File: content/en/blog/2024/sig-api-machinery-spotlight.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+   were still heavily focused on in-tree types.  We built the watch cache, protobuf serialization,
+   and shared caches.
+
+3. **Fostering the ecosystem**: `v1.8-1.21` (up to CRD v1) or so.  This when we designed and wrote
```
Comment (kaslin) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:116]: This when -> This was when

File: content/en/blog/2024/sig-api-machinery-spotlight.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+features that reduce the blast radius for accidents (webhook match conditions) and which provide
+alternative mechanisms with lower risk profiles for most actions (validating admission policy).
+
+At the same time, the increased usage has made us more aware of scaling limitations that we can
+improve both server and client-side.  Efforts here include more efficient serialization (CBOR),
+reduced etcd load (consistent reads from cache), and reduced peak memory usage (streaming lists).
+
+And finally, the increased usage has highlighted some long existing gaps that we’re closing.  Things
+like field selectors for CRDs which wg-batch is eager to leverage and will eventually form the basis
+for a new way to prevent trampoline pod attacks from exploited nodes.
```
Comment (kaslin) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:152]: Would love to know what a "trampoline pod attack" is.

File: content/en/blog/2024/sig-api-machinery-spotlight.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+**Frederico (FSM)**: Hello, and thank your time. To start with, could you tell us about
+yourselvesand how you got involved in Kubernetes?
+
+**David**: I started working on
+[OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift) (the Red Hat
+distribution of Kubernetes) in the fall of 2014 and got involved pretty quickly in apimachinery.  My
+first PRs were fixing kube-apiserver error messages and from there I branched out to `kubectl`
+(_kubeconfigs_ are my fault!), `auth` ([RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) and `*Review` APIs are ports
+from OpenShift), `apps` (_workqueues_ and _sharedinformers_ for example).  Don’t tell the others,
+but apimachinery is still my favorite :)
```
Comment (kaslin) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:24]: apimachinery -> API Machinery

**Review by kaslin — COMMENTED** [review]
Made a few minor nit-pick comments.

File: content/en/blog/2024/sig-api-machinery-spotlight.md (hunk 5 above)
Comment (fsmunoz) [reply content/en/blog/2024/sig-api-machinery-spotlight.md:152]: Maybe we can add https://security.googleblog.com/2022/05/privileged-pod-escalations-in.html , either as a link or a an editor's note?

**Comment by fsmunoz** [comment]
@kaslin I've reflected all changes that I've marked resolved in the last commit, thanks!

File: content/en/blog/2024/sig-api-machinery-spotlight.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+author: "Frederico Muñoz (SAS Institute)"
+---
+
+We recently talked with [Federico Bongiovanni](https://github.com/fedebongio) (Google) and [David
+Eads](https://github.com/deads2k) (Red Hat), Chairs of SIG API Machinery, to know a bit more about
+this Kubernetes Special Interest Group.
+
+## Introductions
+
+**Frederico (FSM): Hello, and thank your time. To start with, could you tell us about
```
Comment (deloitte-saghili) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:15]: little typo: thank you for your time

Comment (aghilish) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:15]: little typo: thank you for your time

Comment (Princesso) [reply content/en/blog/2024/sig-api-machinery-spotlight.md:15]: Apart from this minor fix, it looks good to me.

File: content/en/blog/2024/sig-api-machinery-spotlight.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+yourselves and how you got involved in Kubernetes?**
+
+**David**: I started working on
+[OpenShift](https://www.redhat.com/en/technologies/cloud-computing/openshift) (the Red Hat
+distribution of Kubernetes) in the fall of 2014 and got involved pretty quickly in API Machinery.  My
+first PRs were fixing kube-apiserver error messages and from there I branched out to `kubectl`
+(_kubeconfigs_ are my fault!), `auth` ([RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) and `*Review` APIs are ports
+from OpenShift), `apps` (_workqueues_ and _sharedinformers_ for example).  Don’t tell the others,
```
Comment (ArvindParekh) [inline content/en/blog/2024/sig-api-machinery-spotlight.md:23]:
```suggestion
distribution of Kubernetes) in the fall of 2014 and got involved pretty quickly in API Machinery. My
first PRs were fixing kube-apiserver error messages and from there I branched out to `kubectl`
(_kubeconfigs_ are my fault!), `auth` ([RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) and `*Review` APIs are ports
from OpenShift), `apps` (_workqueues_ and _sharedinformers_ for example). Don’t tell the others,
```

I have noticed extra spaces here, but those doesn't seem to affect the formatting in the final version, so it's fine to let it be.

**Review by ArvindParekh — COMMENTED** [review]
Looks good!

File: content/en/blog/2024/sig-api-machinery-spotlight.md (hunk 7 above)
Comment (fsmunoz) [reply content/en/blog/2024/sig-api-machinery-spotlight.md:15]: ty, fixed in last commit.

**Comment by fsmunoz** [comment]
Updated publication date, in sync with https://github.com/kubernetes/website/pull/47147 .

**Comment by sftim** [comment]
/hold

This needs:
- assigned **future** publication date
- LGTM from a reviewer
- ideally also LGTM on the associated K8s blog PR

OK to unhold once we have those

**Comment by fsmunoz** [comment]
@sftim updated publication date for the 24th, will wait for the LGTM.

**Comment by fsmunoz** [comment]
Updated publication date for next week.

**Pull request merged**